tests/backup/backup_test.go

```diff
@@ -1129,6 +1129,214 @@ var _ = Describe("{ShareBackupAndEdit}", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range bkpNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					time.Sleep(3 * time.Second)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{namespace},
+							nil, orgID, clusterUid, "", "", "", "", ctx)
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+		Step("Share backup with multiple users", func() {
+			log.InfoD("Share backup with multiple users")
+			log.Infof("Sharing backup with  multiple users")
+
+			// Get Admin Context - needed to share backup and get backup UID
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			// Share backups with aal the users
+			for _, backup := range backupNames {
+				err = ShareBackup(backup, nil, users, ViewOnlyAccess, ctx)
+				log.FailOnError(err, "Failed to share backup %s", backup)
+			}
+
+			for _, user := range users {
+				// Get user context
+				ctxNonAdmin, err := backup.GetNonAdminCtx(user, "Password1")
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				userContexts = append(userContexts, ctxNonAdmin)
+
+				for _, backup := range backupNames {
+					// Get Backup UID
+					backupDriver := Inst().Backup
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+
+					// Delete backup to confirm that the user cannot delete the backup
+					_, err = DeleteBackup(backup, backupUID, orgID, ctxNonAdmin)
+					log.Infof("Error message - %s", err.Error())
+					dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
+
+				}
+			}
+			// Delete the backups
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			backupDriver := Inst().Backup
+			for _, backup := range backupNames {
+				wg.Add(1)
+				go func(backup string) {
+					defer wg.Done()
+					backupUID, err := backupDriver.GetBackupUID(ctx, backup, orgID)
+					log.FailOnError(err, "Failed while trying to get backup UID for - %s", backup)
+					_, err = DeleteBackup(backup, backupUID, orgID, ctx)
+					log.FailOnError(err, "Failed to delete backup - %s", backup)
+					err = backupDriver.WaitForBackupDeletion(ctx, backup, orgID, time.Minute*10, time.Minute*1)
+					log.FailOnError(err, "Error waiting for backup deletion %v", backup)
+				}(backup)
+			}
+			wg.Wait()
+		})
```

Comment:
Missed the step where we need to validate that the users who had view only access cannot enumerate the backups after deletion

Reply:
Done